src/subscription_manager/cli_command/remove.py

```diff
@@ -0,0 +1,176 @@
+                print(_("The entitlement server failed to remove these serial numbers:"))
+            else:
+                print(_("The entitlement server failed to remove these IDs:"))
+            for id_ in failure:
+                print("   %s" % id_)
+
+    def _do_command(self):
+        """
+        Executes the command.
+        """
+        self._validate_options()
+        return_code = 0
+        if self.is_registered():
+            ent_service = entitlement.EntitlementService(self.cp)
+            try:
+                if self.options.all:
+                    total = ent_service.remove_all_entitlements()
+                    # total will be None on older Candlepins that don't
+                    # support returning the number of subscriptions unsubscribed from
+                    if total is None:
+                        print(_("All subscriptions have been removed at the server."))
+                    else:
+                        count = total['deletedRecords']
+                        print(ungettext("%s subscription removed at the server.",
+                                        "%s subscriptions removed at the server.",
+                                        count) % count)
+                else:
+                    # Try to remove subscriptions defined by pool IDs first (remove --pool=...)
+                    if self.options.pool_ids:
+                        removed_pools, unremoved_pools, removed_serials = ent_service.remove_entilements_by_pool_ids(
+                            self.options.pool_ids)
+                        if not removed_pools:
+                            return_code = 1
+                        self._print_unbind_ids_result(removed_pools, unremoved_pools, "pools")
+                    else:
+                        removed_serials = []
+                    # Then try to remove subscriptions defined by serials (remove --serial=...)
+                    unremoved_serials = []
+                    if self.options.serials:
+                        serials = unique_list_items(self.options.serials)
+                        # Don't remove serials already removed by a pool
+                        serials_to_remove = [serial for serial in serials if serial not in removed_serials]
+                        _removed_serials, unremoved_serials = ent_service.remove_entitlements_by_serials(
+                            serials_to_remove)
+                        removed_serials.extend(_removed_serials)
+                        if not _removed_serials:
+                            return_code = 1
+                    # Print final result of removing pools
+                    self._print_unbind_ids_result(removed_serials, unremoved_serials, "serial numbers")
+            except connection.GoneException as ge:
+                raise ge
+            except connection.RestlibException as err:
+                log.error(err)
+                system_exit(os.EX_SOFTWARE, err.msg)
```

Comment:
What version of Python is this running on?  Python 3 removes the Exception message, it would just be contained in the variable used in the 'as' part of the statement